src/plugins/job.hpp

```diff
@@ -0,0 +1,56 @@
+/************************************************************\
+ * Copyright 2025 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+// header file for the Job class
+extern "C" {
+#if HAVE_CONFIG_H
+#include "config.h"
+#endif
+#include <flux/core.h>
+#include <flux/jobtap.h>
+#include <jansson.h>
+}
+
+#ifndef JOB_H
+#define JOB_H
+
+#include <vector>
+#include <string>
+#include <map>
+#include <iterator>
+#include <sstream>
+#include <algorithm>
+
+// custom job resource counting file
+#include "jj.hpp"
+
+class Job {
+public:
+    // attributes
+    long int id = 0;               // the ID of the job
+    std::vector<std::string> deps; // any dependencies on job
+    int nnodes = 0;                // the number of nodes requested
+    int ncores = 0;                // the number of cores requested
+    std::string queue;             // the queue the job was submitted under
+
+    // constructor
+    Job () = default;
+};
+
+// count the resources requested for a job
+int count_resources (Job &job, json_t *jobspec);
+
+// determine if a job contains a certain dependency
+bool contains_dep (const Job &job, const std::string &dep);
+
+// remove a job dependency from a job's list of dependencies
+void remove_dep (Job &job, const std::string &dep);
```

Comment:
I'm not the greatest C++ developer, but could these functions should be public members of the Job class, instead of functions that take a `Job` object?

Also, should there be an `add_dep()` method as well as `remove_dep()`? Then perhaps the `deps` vector could be private.

Reply:
Those are great suggestions, thanks! I'll go ahead and change these just to be members of the `Job` class. I'll also go ahead and add an `add_dep ()` method too.

I think there might be a use case or two where the plugin might benefit from being able to access both the size of the `deps` vector as well as seeing the order in which dependencies are placed onto this vector. Would it be okay if I left this public?

Reply:
Oh yeah, that makes sense!